src/exchange/passthrough.go

```diff
@@ -254,8 +256,56 @@ func (p *Passthrough) getCheapestAsk(di DepositInfo) (*exchange.MarketOrder, err
 }
 
 // placeOrder places an order on the exchange and returns the orderID
-func (p *Passthrough) placeOrder(di DepositInfo, ask *exchange.MarketOrder) (int, error) {
-	return p.exchangeClient.Buy(fmt.Sprintf("SKY_%s", di.CoinType), ask.Price, ask.Volume)
+func (p *Passthrough) placeOrder(di DepositInfo, ask *exchange.MarketOrder) (int, *exchange.MarketOrder, error) {
+	order, err := buildOrderFromDeposit(di, ask)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	orderID, err := p.exchangeClient.Buy(fmt.Sprintf("SKY_%s", di.CoinType), order.Price, order.Volume)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	return orderID, order, nil
+}
+
+// buildOrderFromDeposit creates an order given a deposit and an ask order.
+// The order's volume will based upon the amount of BTC to spend on the order,
+// given the fixed price of the input ask order.
+func buildOrderFromDeposit(di DepositInfo, ask *exchange.MarketOrder) (*exchange.MarketOrder, error) {
+	// Calculate the cost of the ask order in BTC
+	// If the amount available to spend is less than the cost of the order,
+	// adjust the volume of the ask order to match it.
+	if di.DepositValue <= di.Passthrough.DepositValueSpent {
+		return nil, errors.New("deposit has been entirely spent")
+	}
+
+	cost := askCost(ask)
+	available := decimal.New(di.DepositValue-di.Passthrough.DepositValueSpent, -int32(SatoshiExponent))
+
+	// We can afford the entire order, return it unmodified
+	if available.GreaterThan(cost) {
+		return ask, nil
+	}
+
+	// We will bid on part of the order, adjust the volume downward
+	volume := available.Div(ask.Price)
+
+	// Truncate the volume to the maximum decimals spendable by Skycoin
+	volume = volume.Truncate(int32(visor.MaxDropletPrecision))
+
+	return &exchange.MarketOrder{
+		Price:  ask.Price,
+		Volume: volume,
+	}, nil
+}
+
+// Calculates the cost of an ask order
+// TODO -- make this a method of exchange.MarketOrder
+// exchange.MarketOrder should also specify the currency types involved
+func askCost(ask *exchange.MarketOrder) decimal.Decimal {
+	return ask.Price.Mul(ask.Volume)
```

Comment:
c2cx supports only like 4 decimals from what I can see on the site so shouldn't we have a truncation here as well?

Reply:
Waiting for a reply from their support

Reply:
Minimum volume: 1 SKY
Volume decimal places: 2
Price decimal places: 5

These need to be hardcoded as constants in the exchange-api library